wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Poll/Editor.ts

```diff
@@ -0,0 +1,404 @@
+  private readonly questionField: HTMLInputElement;
+  private readonly resultsRequireVoteNoField: HTMLInputElement;
+  private readonly resultsRequireVoteYesField: HTMLInputElement;
+  private readonly sortByVotesNoField: HTMLInputElement;
+  private readonly sortByVotesYesField: HTMLInputElement;
+  private readonly wysiwygId: string;
+  
+  constructor(containerId: string, pollOptions: PollOption[], wysiwygId: string, options: UiPollEditorOptions) {
+    const container = document.getElementById(containerId);
+    if (container === null) {
+      throw new Error("Unknown poll editor container with id '" + containerId + "'.");
+    }
+    this.container = container;
+
+    this.wysiwygId = wysiwygId;
+    if (wysiwygId !== '' && document.getElementById(wysiwygId) === null) {
+      throw new Error("Unknown wysiwyg field with id '" + wysiwygId + "'.");
+    }
+
+    this.questionField = document.getElementById(this.wysiwygId + 'Poll_question') as HTMLInputElement;
+
+    const optionList = this.container.querySelector('.sortableList');
+    if (optionList === null) {
+      throw new Error("Cannot find poll options list for container with id '" + containerId + "'.");
+    }
+    this.optionList = optionList as HTMLOListElement;
+
+    this.endTimeField = document.getElementById(this.wysiwygId + 'Poll_endTime') as HTMLInputElement;
+    this.maxVotesField = document.getElementById(this.wysiwygId + 'Poll_maxVotes') as HTMLInputElement;
+    this.isChangeableYesField = document.getElementById(this.wysiwygId + 'Poll_isChangeable') as HTMLInputElement;
+    this.isChangeableNoField = document.getElementById(this.wysiwygId + 'Poll_isChangeable_no') as HTMLInputElement;
+    this.isPublicYesField = document.getElementById(this.wysiwygId + 'Poll_isPublic') as HTMLInputElement;
+    this.isPublicNoField = document.getElementById(this.wysiwygId + 'Poll_isPublic_no') as HTMLInputElement;
+    this.resultsRequireVoteYesField = document.getElementById(this.wysiwygId + 'Poll_resultsRequireVote') as HTMLInputElement;
+    this.resultsRequireVoteNoField = document.getElementById(this.wysiwygId + 'Poll_resultsRequireVote_no') as HTMLInputElement;
+    this.sortByVotesYesField = document.getElementById(this.wysiwygId + 'Poll_sortByVotes') as HTMLInputElement;
+    this.sortByVotesNoField = document.getElementById(this.wysiwygId + 'Poll_sortByVotes_no') as HTMLInputElement;
+    
+    this.optionCount = 0;
+    
+    this.options = Core.extend({
+      isAjax: false,
+      maxOptions: 20,
+    }, options) as UiPollEditorOptions;
+
+    this.createOptionList(pollOptions || []);
+
+    new UiSortableList({
+      containerId: containerId,
+      options: {
+        toleranceElement: '> div',
+      },
+    });
+    
+    if (this.options.isAjax) {
+      ['handleError', 'reset', 'submit', 'validate'].forEach((event) => {
+        EventHandler.add(
+          'com.woltlab.wcf.redactor2',
+          event + '_' + this.wysiwygId,
+          this[event].bind(this)
```

Comment:
I'm not overly happy with this, but I don't think it gets any more type safe:
```suggestion
          (...args: unknown[]) => this[event](...args)
```
Possibly also get rid of the loop and make this explicit.